identity/src/secp256k1.rs

```diff
@@ -193,31 +190,46 @@ impl cmp::Ord for PublicKey {
 impl PublicKey {
     /// Verify the Secp256k1 signature on a message using the public key.
     pub fn verify(&self, msg: &[u8], sig: &[u8]) -> bool {
-        self.verify_hash(Sha256::digest(msg).as_ref(), sig)
+        let digest = Sha256::new_with_prefix(msg);
+        self.verify_hash(digest.finalize().as_slice(), sig)
     }
 
-    /// Verify the Secp256k1 DER-encoded signature on a raw 256-bit message using the public key.
+    /// Verify the Secp256k1 DER-encoded signature on a raw 256-bit message using the public key.  
+    /// Will panic if the hash is not 32 bytes long.
     pub fn verify_hash(&self, msg: &[u8], sig: &[u8]) -> bool {
-        Message::parse_slice(msg)
-            .and_then(|m| Signature::parse_der(sig).map(|s| libsecp256k1::verify(&m, &s, &self.0)))
-            .unwrap_or(false)
+        Signature::from_der(sig).is_ok_and(|s| {
+            k256::ecdsa::hazmat::verify_prehashed(
+                &ProjectivePoint::from(self.0.as_affine()),
+                GenericArray::from_slice(msg),
```

Comment:
This introduces `generic_array` as a direct dependency.

Reply:
It doesn't seem necessary, see d197c8bd0fbfc1382bb25ae6397d364c27455a70